marathon_lb.py

```diff
@@ -1891,21 +1891,44 @@ def load_json(data_str):
                                            args.haproxy_map)
         signal.signal(signal.SIGHUP, processor.handle_signal)
         signal.signal(signal.SIGUSR1, processor.handle_signal)
-        backoff = 3
+        backoffFactor = 1.5
+        waitSeconds = 3
+        maxWaitSeconds = 300
+        waitResetSeconds = 600
         while True:
             stream_started = time.time()
+            currentWaitSeconds = random.random() * waitSeconds
             try:
                 process_sse_events(marathon, processor)
+            except pycurl.error, e:
+                errno, e_msg = e.args
+                # Error number 28:
+                # 'Operation too slow. Less than 1 bytes/sec transferred
+                #  the last 300 seconds'
+                # In this case we should immediately reconnect
+                # without a backoff.
+                if errno == 28:
+                    m = 'Possible timeout detected: {}, reconnecting now...'
+                    logger.info(m.format(e_msg))
+                    currentWaitSeconds = 0
+                else:
+                    raise
```

Comment:
does this `raise` trigger a reconnect? I'm not familiar with python exception handling

Reply:
Great catch! This would actually exit the loop completely and not get caught by the following `except`. The intended behavior would be to log the exception and then reconnect with backoff as normal.